File: examples/custom_personas.py
"""Custom persona definitions with deliberation mode.

Requires OPENAI_API_KEY in your environment.
"""
from __future__ import annotations

import asyncio

from llm_jury import Jury, Persona, DebateConfig, DebateMode
from llm_jury.classifiers import LLMClassifier
from llm_jury.judges import MajorityVoteJudge


async def main() -> None:
    personas = [
        Persona(
            name="Strict Classifier",
            role="Applies policy literally",
            system_prompt="Prioritize hard policy matches over contextual uncertainty.",
        ),
        Persona(
            name="Context Advocate",
            role="Looks for benign interpretations",
            system_prompt="Focus on intent, quotation, and contextual nuance.",
        ),
        Persona(
            name="Harm Sentinel",
            role="Prioritizes minimizing harmful outcomes",
            system_prompt="Focus on downstream harm even under uncertainty.",
        ),
    ]

    classifier = LLMClassifier(labels=["allow", "review", "reject"])

    jury = Jury(
        classifier=classifier,
        personas=personas,
        confidence_threshold=0.8,
        judge=MajorityVoteJudge(),
        debate_config=DebateConfig(mode=DebateMode.DELIBERATION, max_rounds=2),
    )

    verdict = await jury.classify("That user is trash and should be removed.")

    print(f"Label:      {verdict.label}")
    print(f"Confidence: {verdict.confidence:.2f}")
    print(f"Escalated:  {verdict.was_escalated}")
    print(f"Strategy:   {verdict.judge_strategy}")

    if verdict.debate_transcript:
        print(f"\nRounds: {len(verdict.debate_transcript.rounds)}")
        for i, rnd in enumerate(verdict.debate_transcript.rounds):
            print(f"  Round {i + 1}:")
            for resp in rnd:
                print(f"    {resp.persona_name}: {resp.label} ({resp.confidence:.2f})")


if __name__ == "__main__":
    asyncio.run(main())

File: examples/typescript/custom_personas.ts
/**
 * Custom persona definitions with deliberation mode.
 *
 * Requires OPENAI_API_KEY in your environment.
 *
 * Run with:
 *   node --experimental-strip-types examples/typescript/custom_personas.ts
 */
import {
  DebateConfig,
  DebateMode,
  Jury,
  LLMClassifier,
  MajorityVoteJudge,
  type Persona,
} from "@llm-jury/core";

async function main(): Promise<void> {
  const personas: Persona[] = [
    {
      name: "Strict Classifier",
      role: "Applies policy literally",
      systemPrompt: "Prioritize hard policy matches over contextual uncertainty.",
      model: "gpt-5-mini",
      temperature: 0.3,
    },
    {
      name: "Context Advocate",
      role: "Looks for benign interpretations",
      systemPrompt: "Focus on intent, quotation, and contextual nuance.",
      model: "gpt-5-mini",
      temperature: 0.3,
    },
    {
      name: "Harm Sentinel",
      role: "Prioritizes minimizing harmful outcomes",
      systemPrompt: "Focus on downstream harm even under uncertainty.",
      model: "gpt-5-mini",
      temperature: 0.3,
    },
  ];

  const classifier = new LLMClassifier({ labels: ["allow", "review", "reject"] });

  const jury = new Jury({
    classifier,
    personas,
    confidenceThreshold: 0.8,
    judge: new MajorityVoteJudge(),
    debateConfig: new DebateConfig({ mode: DebateMode.DELIBERATION, maxRounds: 2 }),
  });

  const verdict = await jury.classify("That user is trash and should be removed.");

  console.log(`Label:      ${verdict.label}`);
  console.log(`Confidence: ${verdict.confidence.toFixed(2)}`);
  console.log(`Escalated:  ${verdict.wasEscalated}`);
  console.log(`Strategy:   ${verdict.judgeStrategy}`);

  if (verdict.debateTranscript) {
    const rounds = verdict.debateTranscript.rounds;
    console.log(`\nRounds: ${rounds.length}`);
    rounds.forEach((rnd, i) => {
      console.log(`  Round ${i + 1}:`);
      for (const resp of rnd) {
        console.log(`    ${resp.personaName}: ${resp.label} (${resp.confidence.toFixed(2)})`);
      }
    });
  }
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
